galaxy_api/api/v3/viewsets.py

```diff
@@ -161,5 +164,27 @@ def _prepare_post_params(data):
 
 
 class CollectionArtifactDownloadView(views.APIView):
-    def download(self, request, *args, **kwargs):
-        pass
+    def get(self, request, *args, **kwargs):
+        # NOTE(cutwater): Using urllib3 because it's already a dependency of pulp_galaxy
+        url = 'http://{host}:{port}/{prefix}/automation-hub/{filename}'.format(
+            host=settings.PULP_CONTENT_HOST,
+            port=settings.PULP_CONTENT_PORT,
+            prefix=settings.PULP_CONTENT_PATH_PREFIX.strip('/'),
+            filename=self.kwargs['filename'],
+        )
+        response = requests.get(url, stream=True, allow_redirects=False)
```

Comment:
Yes, I know about that approach. But since only limited number of status codes (including errors) is expected, anything outside is treated as an error. Code that handles `raise_for_status` looks little uglier.